src/App.tsx

```diff
@@ -10,13 +10,18 @@ import Register from "./pages/Register";
 import { ProtectedRoute } from "./components/Auth/ProtectedRoute";
 import PageLoader from "./pages/PageLoader";
 import Profile from "./pages/Profile";
-import { useUserContext } from "./context/UserContext";
 import "./App.css";
+import { RootState } from "./state/store";
+import { useDispatch, useSelector } from "react-redux";
+import { setUser } from "./state/slices/userSlice";
 import { PostContextProvider } from "./context/PostContext";
 
 function App() {
   const { isLoading, isAuthenticated, getAccessTokenSilently } = useAuth0();
-  const { user, setUser } = useUserContext();
+  // const { user, setUser } = useUserContext();
+
+  const user = useSelector((state: RootState) => state.user.value);
```

Comment:
If we get rid of the nested `value` property like I suggest in `userSlice.ts` we can do something like this instead:

```suggestion
  const { isLoading, userId }= useSelector((state: RootState) => state.user);
```

I'd also like to create a custom hook that warps `useSelector` so that we don't have to keep passing the `state: RootState` type